src/test/java/com/Incubatormsrv/Incubatormsrv/IncubatorMsrvApplicationTests.java

```diff
@@ -1,13 +1,204 @@
 package com.Incubatormsrv.Incubatormsrv;
 
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.api.WeatherApi;
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.model.CurrentWeatherResponse;
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.model.WeatherResponse;
+import com.Incubatormsrv.Incubatormsrv.CurrentWeather.CurrentWeatherController;
+import com.Incubatormsrv.Incubatormsrv.weather.ThreadExecutor;
+import com.Incubatormsrv.Incubatormsrv.weather.WeatherApiException;
+import com.Incubatormsrv.Incubatormsrv.weather.mapper.WeatherMapper;
+import com.Incubatormsrv.Incubatormsrv.weather.service.WeatherService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.google.gson.Gson;
+import jakarta.servlet.http.HttpServletRequest;
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import retrofit2.Call;
+import retrofit2.Response;
 
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.doReturn;
+
+@ActiveProfiles("test")
+@RunWith(MockitoJUnitRunner.class)
 @SpringBootTest
 class IncubatorMsrvApplicationTests {
+	@Mock
+	ThreadExecutor threadExecutor;
+	@Mock
+	CurrentWeatherController currentWeatherController;
+	@Mock
+	WireMockServer wireMock;
+	@InjectMocks
+	private static WeatherMapper weatherMapper;
+	static WeatherService weatherService;
+	@BeforeAll
+	static void setup(){
```

Comment:
You must clear your wifemock server else you could have a case of where an old stub could cause your test to be invalid,